docs/proposal-005-upgrades.md

```diff
@@ -0,0 +1,106 @@
+# Design Proposal #005: Upgrades
+
+> **STATUS**: This proposal is a _final_ state, and we expect minimal additional refinements.
+> If any non-trivial changes are needed to functionality defined here, in particular the user
+> experience, those changes should be suggested via a PR to this proposal document.
+> Any other changes to the text of the proposal or technical corrections are also very welcome.
+
+With `eksctl` users should be able to easily upgrade from one version of Kubernetes to another.
+
+Cluster upgrades are inherently a multi-step process, and can be fairly complex. It's important
+that user can drive upgrades at their own pace, if they must.
+
+There is a standard set of steps a user needs to take, most of which can be automated as a single
+command also. However, there maybe additional manual steps with some versions, however most parts
+should be automated.
+
+When upgrading a cluster, one needs to call `eks.UpdateClusterVersion`. In some cases other changes
+to cluster stack maybe required. After that they need to replace nodegroups one by one. 
+
+## Initial phase
+
+- provide command that checks cluster stack for upgradability
+  - let's user update cluster stack to cater for any additional resources
+  - allows to call `eks.UpdateClusterVersion` out-of-band and wait for completion
+- provide instruction on how to iteratively replace nodegroups
+- provide helper commands for upgrading add-ons
+
+## Second phase (more automation)
+
+- use CloudFormation instead of calling `eks.UpdateClusterVersion` directly
+- provide automated command that replaces all nodegroups by copying configuration,
+  or at least one command for each nodegroup
+
+## CLI Design
+
+For initial phase:
+
+Update cluster version and append new resources to the cluster stack (if needed):
+```
+eksclt update cluster --name=<clusterName>
+```
+
+As this can have significant implications and some checks are appropriate before
+we try updating, `--dry-run` should be enabled by default.
+
+As control plane version can be incremented only one at a time, it would be a good
+idea to have `--version` flag. The default should be `--version=next`. It should
+possible to jump many versions with `--version=<v>`, and go for latest right away
+with `--version=latest`. User will have to make sure that this doesn't take them
+too far ahead, so their nodegroup(s) don't drift too far behind.
+
+For each `<currentNodeGroup>`, create `<replacementNodeGroup>`:
+```
+eksctl create ng --cluster=<clusterName> --name=<replacementNodeGroup>
+eksctl delete ng --cluster=<clusterName> --name=<currentNodeGroup>
```

Comment:
I would mention the version part here too

Reply:
👍